### PR TITLE
test(x/ecocredit): migrate update class and project tests

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Ref: #893

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/features/msg_update_project_admin.feature [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    Scenario: the project exists
+      When alice attempts to update project admin with project id "C01-001"
+      Then expect no error
+
+    Scenario: the project does not exist
+      When alice attempts to update project admin with project id "C01-002"
+      Then expect the error "could not get project with id C01-002: not found: invalid request"
+
+  Rule: The admin must be the admin of the project
```
Comment (technicallyty) [inline x/ecocredit/server/core/features/msg_update_project_admin.feature:23]:
```suggestion
  Rule: The user must be the admin of the project
```
i think this was referring to the field `admin` but i think `user` would be more natural here

File: x/ecocredit/server/core/features/msg_update_project_admin.feature [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    Background:
+      Given a credit type with abbreviation "C"
+      And a credit class with class id "C01"
+      And a project with project id "C01-001" and admin alice
+
+    Scenario: the admin is the admin of the project
+      When alice attempts to update project admin with project id "C01-001"
+      Then expect no error
+
+    Scenario: the admin is not the admin of the project
```
Comment (technicallyty) [inline x/ecocredit/server/core/features/msg_update_project_admin.feature:34]: same here, i feel like using `admin` is a bit awkward

File: x/ecocredit/server/core/features/msg_update_project_metadata.feature [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      Then expect the error "could not get project with id C01-002: not found: invalid request"
+
+  Rule: The admin must be the admin of the project
+
+    Background:
+      Given a credit type with abbreviation "C"
+      And a credit class with class id "C01"
+      And a project with project id "C01-001" and admin alice
+
+    Scenario: the admin is the admin of the project
```
Comment (technicallyty) [inline x/ecocredit/server/core/features/msg_update_project_metadata.feature:30]: tracking here too in case we decide to change `admin`

**Review by technicallyty — APPROVED** [review]
lgtm just one nit about admin wording

**Review by aleem1314 — APPROVED** [review]
lgtm

File: x/ecocredit/server/core/features/msg_update_project_admin.feature (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/server/core/features/msg_update_project_admin.feature:23]: Good point. I'm leaning towards using the field name, hence the change here, but I also see value in using "user". In some cases the address might not be a "user" but rather a "group" or a multisig and therefore "signer" might make sense too. I'm going to leave this here now but we should discuss when writing guidelines so I've left a comment in the issue. #1242

Comment (ryanchristo) [reply x/ecocredit/server/core/features/msg_update_project_admin.feature:23]: I'm leaning towards using the field name, hence the change here, but I also see value in using "user". In some cases the address might not be a "user" but rather a "group" or a multisig and therefore "signer" might make sense too. I'm going to leave this here now but we should discuss when writing guidelines so I've left a comment in the issue. #1242

**Pull request merged**